src/encoded/static/components/view_controls.js

```diff
@@ -1,24 +1,160 @@
 import PropTypes from 'prop-types';
-import * as encoding from '../libs/query_encoding';
+import _ from 'underscore';
+import url from 'url';
+import QueryString from '../libs/query_string';
 import { svgIcon } from '../libs/svg-icons';
+import * as DropdownButton from '../libs/ui/button';
 import { getIsCartSearch } from './cart';
 import { BatchDownloadActuator, SearchBatchDownloadController } from './batch_download';
-import QueryString from '../libs/query_string';
+
+
+/**
+ * Displays any of the view type bu†tons, wrapping the actual button title.
+ */
+const ViewControlButton = ({ viewType, queryString, children }) => (
+    <a
+        href={`/${viewType.path}/?${queryString}`}
+        role="button"
+        className="btn btn-info btn-sm"
+        id={`search-control-${viewType.type}`}
+        data-test={viewType.type}
+        aria-label={`${viewType.label}`}
+    >
+        {svgIcon(viewType.icon)}
+        {children}
+    </a>
+);
+
+ViewControlButton.propTypes = {
+    /** Single relevant entry from `viewTypeMap` */
+    viewType: PropTypes.object.isRequired,
+    /** Current search query string */
+    queryString: PropTypes.string.isRequired,
+    /** Button title content */
+    children: PropTypes.node.isRequired,
+};
+
+
+/**
+ * Renders the view type button for the search page.
+ */
+const SearchViewButton = ({ viewType, query }) => (
+    <ViewControlButton viewType={viewType} queryString={query.format()}>
+        {viewType.title}
+    </ViewControlButton>
+);
+
+SearchViewButton.propTypes = {
+    /** Single relevant entry from `viewTypeMap` */
+    viewType: PropTypes.object.isRequired,
+    /** Current search query object */
+    query: PropTypes.object.isRequired,
+};
+
+
+/**
+ * Renders the view type button for the report page. This renders a normal button like the search
+ * and matrix views when one "type=" exists in the query string, but renders a dropdown button when
+ * more than one "type=" exists in the query string -- one entry in the dropdown for each "type="
+ * in the query string.
+ */
+const ReportViewButton = ({ viewType, query }, reactContext) => {
+    const typeValues = query.getKeyValues('type');
+    if (typeValues.length === 1) {
+        // One "type=" in the query string so render a regular link to the report page.
+        return (
+            <ViewControlButton viewType={viewType} queryString={query.format()}>
+                {viewType.title}
+            </ViewControlButton>
+        );
+    }
```

Comment:
It looks like you could use the `<SearchViewButton>` here in lines 66-68 but I think you didn't do that because the code is easier to understand if you have one component for each type of view button. You could rename the SearchViewButton to be something more generic and use it here, but I also think it's fine as-is, it was just a thought I had.